src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs

```diff
@@ -0,0 +1,315 @@
+            using (var firstSender = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            using (var secondSender = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                var list = new List<Socket> { firstSender, secondSender };
+                Socket.Select(null, list, null, SelectSuccessTimeout);
+
+                Assert.Equal(2, list.Count);
+                Assert.Equal(firstSender, list[0]);
+                Assert.Equal(secondSender, list[1]);
+            }
+        }
+
+        [Fact]
+        public void SelectWrite_Multiple_Timeout()
+        {
+            using (var firstListener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            using (var secondListener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                firstListener.BindToAnonymousPort(IPAddress.Loopback);
+                firstListener.Listen(1);
+                firstListener.AcceptAsync();
+                
+                secondListener.BindToAnonymousPort(IPAddress.Loopback);
+                secondListener.Listen(1);
+                secondListener.AcceptAsync();
+
+                var list = new List<Socket> { firstListener, secondListener };
+                Socket.Select(null, list, null, SelectTimeout);
+
+                Assert.Equal(0, list.Count);
+            }
+        }
+
+        [Fact]
+        public void SelectWrite_Multiple_Mixed()
+        {
+            using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            using (var sender = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                listener.BindToAnonymousPort(IPAddress.Loopback);
+                listener.Listen(1);
+                listener.AcceptAsync();
+
+                var list = new List<Socket> { listener, sender };
+                Socket.Select(null, list, null, SelectSuccessTimeout);
+
+                Assert.Equal(1, list.Count);
+                Assert.Equal(sender, list[0]);
+            }
+        }
+
+        [Fact]
+        public void Select_Error_NotASocket_Throws()
+        {
+            var list = new List<object> { new object() };
+            Assert.Throws<ArgumentException>(() => Socket.Select(null, null, list, SelectSuccessTimeout));
+        }
+
+        [Fact]
+        public void SelectError_Single_Timeout()
+        {
+            using (var receiver = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                receiver.BindToAnonymousPort(IPAddress.Loopback);
+
+                var list = new List<Socket> { receiver };
+                Socket.Select(null, null, list, SelectTimeout);
+
+                Assert.Equal(0, list.Count);
+            }
+        }
+
+        [Fact]
+        public void SelectError_Multiple_Timeout()
```

Comment:
Can we somehow add a test for selecting with an error list and having something remain in the list (i.e. there is an error)?